graphula-core/src/Graphula.hs

```diff
@@ -617,7 +612,7 @@ attempt maxEdits maxInserts source = loop 0 0
 
 -- | For entities that only have singular 'Dependencies'. It uses data instead
 -- of newtype to match laziness of builtin tuples.
```

Comment:
Just noticed this comment. I guess I shouldn't blindly follow HLint? This laziness point is important?

Reply:
Eh, my opinion on this has evolved. Now I think a singleton tuple should be representationally identical to its parameter. Technically a breaking change, but no one should be relying on the indirection here. We can do a version bump once you change the exception types tho.

Reply:
Cool, I'll nuke the comment before merging.